Tools/debug/CollectSystemInfo.ps1

```diff
@@ -0,0 +1,214 @@
+    try {
+        $uptime = (Get-Date) - (gcim Win32_OperatingSystem).LastBootUpTime
+        $uptime.ToString() | Out-File -FilePath 'WindowsUptime.txt'
+        Write-Host 'Windows uptime collection completed.'
+    } catch {
+        Write-Warning "Failed to collect Windows uptime: $_"
+    }
+}
+
+function Get-ServicesList {
+    try {
+        Get-Service | Select-Object -Property Name, DisplayName, Status, StartType | Export-Csv -Path 'Services.csv' -NoTypeInformation
+    } catch {
+        Write-Warning "Failed to collect list of services: $_"
+    }
+}
+
+function Get-RunningProcesses {
+    try {
+        Get-Process | Select-Object -Property Id, ProcessName, StartTime | Export-Csv -Path 'RunningProcesses.csv' -NoTypeInformation
+    } catch {
+        Write-Warning "Failed to collect list of running processes: $_"
+    }
+}
+
+function Get-InstalledApplications {
+    try {
+        Get-ItemProperty -Path 'HKLM:\Software\Microsoft\Windows\CurrentVersion\Uninstall\*' |
+        Select-Object -Property DisplayName, DisplayVersion, Publisher, InstallDate |
+        Export-Csv -Path 'InstalledApplications.csv' -NoTypeInformation
+    } catch {
+        Write-Warning "Failed to collect list of installed applications: $_"
+    }
+}
+
+function Get-InstalledKBs {
+    try {
+        Get-HotFix | Select-Object -Property Description, HotFixID, InstalledOn | Export-Csv -Path 'InstalledKBs.csv' -NoTypeInformation
+    } catch {
+        Write-Warning "Failed to collect list of installed KBs: $_"
+    }
+}
+
+function Get-WindowsMemoryDump {
+    $dumpFiles = @()
+    $memoryDumpPaths = @("$env:SystemRoot\MEMORY.DMP", "$env:SystemRoot\Minidump")
+
+    foreach ($path in $memoryDumpPaths) {
+        try {
+            $memoryDumpPath = Get-Item -Path $path -ErrorAction Stop
+            if ($memoryDumpPath) {
+                $dumpFiles += $memoryDumpPath.FullName
+            }
+        } catch {
+            Write-Host "Failed to collect $memoryDumpPath file: $_"
+        }
+    }
+
+    return $dumpFiles
+}
+
+function Remove-TempFiles {
+    param ([string[]]$Path)
+    try {
+        Remove-Item -Path $Path -Recurse -Force
+    } catch {
+        Write-Warning "Failed to delete file: $resolvedFile - $_"
+    }
+}
+
+function Write-InformationToArchive {
+    try {
+        $date = Get-Date -Format 'yyyy-MM-dd_HH-mm-ss'
+        $archiveFile = "SystemInfo_$date.zip"
+        $filesInCurrentDirectory = @('*.txt', '*.csv', '*.evtx', '*.log')
+        $collectedFiles = foreach ($pattern in $filesInCurrentDirectory) {
+            Get-ChildItem -Path $pattern -ErrorAction SilentlyContinue
+        }
+
+        if (Test-Path -Path 'LocaleMetaData') {
+            $collectedFiles += 'LocaleMetaData'
+        }
+
+        if ($IncludeSensitiveData) {
+            $dumpFiles = Get-WindowsMemoryDump
+            $exportedFiles += $dumpFiles
+        }
+        
+        $filesToArchive = $collectedFiles + $exportedFiles | Where-Object { $_ }
```

Comment:
What is `Where-Object { $_ }` for?

Reply:
Why does it have null objects?

Reply:
> Why does it have null objects?
It's not necessary that it contains null object, I added it for preventing issues

Reply:
We shouldn't do that.
- There are so many places where we can have similar filtering code. Adding filtering code for all of them makes the codebase too cluttered. There is also no reason to add such code just only here.
- Raising an exception when dereferencing null is not a flaw, but a feature of Powershell, so that behavior should be preserved. Another language design option is to make all expressions involving null evaluated as null, but it is not adopted by Powershell (and other common languages) because it hides programming errors.